### PR TITLE
[Bugfix] - Fixed adding weapons to a gravestone.

### PR DESCRIPTION
**Pull request by Bubbus**
Fixes this:
image

### Pull Request Description
**When merged this pull request will:**
- Fix adding weapons to a gravestone.
- Weapons in gravestones keep their attachments attached.
- Backpack-gravestones no longer contain duplicate backpacks.
- Banish the Syndikat Crate™️ which appears when the mishmaker deletes all the gravestones.

### Release Notes
- Fixed adding weapons to a gravestone.
- Weapons in gravestones keep their attachments attached.
- Backpack-gravestones no longer contain duplicate backpacks.
- Banished the Syndikat Crate™️ which appears when the mishmaker deletes all the gravestones.

## IMPORTANT

- [x] Testing has been completed as neccessary, depending on the nature & impact of the changes.
- [x] The Release Notes section below must be filled out appropriately to explain the changes made in this PR. This section will be used in Framework Changelogs.
- [x] If the contribution affects [the wiki](https://github.com/CombinedArmsGaming/CAFE3/wiki), please include your changes in this pull request so the wiki is consistently updated.
- [x] [Contribution Guidelines](https://github.com/CombinedArmsGaming/CAFE3/blob/release/contributing.md) are read, understood and applied.
- [x] Title of this PR uses our standard template `[Descriptor] - Add|Fix|Improve|Change|Make|Remove {changes}`.

### CODE REVIEW

File: components/miscShared/functions.hpp
Comment (Cre8or) [inline components/miscShared/functions.hpp:1]: Looks like git is freaking out over this file. I imagine the order of some functions has changed? It's difficult to see the "real" diff.

File: components/miscShared/fn_insertLoadoutIntoCrate.sqf [hunk 1]
```diff
@@ -11,44 +11,21 @@ clearItemCargoGlobal _crate;
...
-        _bipod = _array select 6;
-    };
-
-    _weapons pushBack [_weapon, 1];
-    if (count _primaryMag > 0) then {_magazines pushBack _primaryMag};
-    if (count _secondaryMag > 0) then {_magazines pushBack _secondaryMag};
-    _items append ([_muzzleDevice, _pointer, _optic, _bipod] call BIS_fnc_consolidateArray);
-
+    _weapons pushBack _array;
 };
```
Comment (Cre8or) [inline components/miscShared/fn_insertLoadoutIntoCrate.sqf]: Perhaps we could get rid of this function? We currently still call it with the full 4 arguments, despite only needing 2 in order to do ` _weapons pushBack _array`. Instead we could simply replace the function calls with that line - improves readability, and reduces the amount of script calls.

**Review by Cre8or — CHANGES_REQUESTED** [review]
Tested in singleplayer, works as expected. I only have one nitpick, other than that it's good.

Also ty for retroactively making many variables `private` - it's tech debt but good to see we're slowly addressing it. 👍

**Review by Joecuronium — COMMENTED** [review]
Apart from cre's changes i implemented this in sundays op and it worked without a single hiccup

File: components/miscShared/fn_insertLoadoutIntoCrate.sqf (hunk 1 above)
Comment (Bubbus) [reply components/miscShared/fn_insertLoadoutIntoCrate.sqf]: Done!

File: components/miscShared/functions.hpp
Comment (Bubbus) [reply components/miscShared/functions.hpp:1]: I alphabetised this file, yeah the diff is a mess.

File: components/gravestones/init_component.sqf
Comment (Cre8or) [inline components/gravestones/init_component.sqf:1]: Good bye little box - you won't be missed. ❤️

**Review by Cre8or — APPROVED** [review]
All clear, happy sailing.

🚢

**Pull request merged**